pkg/v1/remote/transport/ping.go

```diff
@@ -80,17 +80,19 @@ func ping(ctx context.Context, reg name.Registry, t http.RoundTripper) (*pingRes
 	}
 
 	var errs []string
-	for _, scheme := range schemes {
+	for i, scheme := range schemes {
 		url := fmt.Sprintf("%s://%s/v2/", scheme, reg.Name())
 		req, err := http.NewRequest(http.MethodGet, url, nil)
 		if err != nil {
 			return nil, err
 		}
 		resp, err := client.Do(req.WithContext(ctx))
 		if err != nil {
-			errs = append(errs, err.Error())
 			// Potentially retry with http.
-			continue
+			if i < len(schemes)-1 {
```

Comment:
I think we want the dumb multi-err thing (maybe in a less dumb way) because it's really confusing otherwise.